sdk/daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/UpgradeTest.scala

```diff
@@ -1415,6 +1568,23 @@ class UpgradeTest extends AnyFreeSpec with Matchers {
           None // There are no fetch* or lookupByKey commands
         case (_, Exercise | ExerciseInterface, _, Command, Local) =>
           None // Local contracts cannot be exercised by commands, except by key
+        case (
+              AdditionalFieldInChoiceArgRecordField | AdditionalConstructorInChoiceArgVariantField |
+              AdditionalConstructorInChoiceArgEnumField | AdditionalChoiceArg |
+              ValidDowngradeAdditionalFieldInChoiceArgRecordField |
+              InvalidDowngradeAdditionalFieldInChoiceArgRecordField |
+              ValidDowngradeAdditionalConstructorInChoiceArgVariantField |
+              InvalidDowngradeAdditionalConstructorInChoiceArgVariantField |
+              ValidDowngradeAdditionalConstructorInChoiceArgEnumField |
+              InvalidDowngradeAdditionalConstructorInChoiceArgEnumField |
+              ValidDowngradeAdditionalChoiceArg | InvalidDowngradeAdditionalChoiceArg,
```

Comment:
This is quite a pattern, could this be a property on the test case?

Reply:
I've gone back and forth with adding it as a property on the test case but then I'm not sure where to draw the line. I could also have "supportsCommand" and similar conditions for other patterns. I tried that first and it was a bit hard to figure out where the whitelist/blacklists were specified. In the end I think that it's easier if it's all in one place, even if it's a bit verbose.